install/0000_30_machine-api-operator_04_metal3provisioning.crd.yaml

```diff
@@ -81,6 +81,24 @@ spec:
               description: provisioningOSDownloadURL is the location from which the OS Image used to boot
                 baremetal host machines can be downloaded by the metal3 cluster.
               type: string
+            provisioningNetwork:
+              description: provisioningNetwork provides a way to indicate the state of the underlying
+                network configuration for the provisioning network. This field can have one of the
+                following values -
+                `managed`- when the provisioning network is completely managed by the Baremetal IPI
+                solution.
+                `unmanaged`- when the provsioning network is present and used but the user is
+                responsible for managing DHCP. Virtual media provisioning is recommended but PXE
+                is still available if required.
+                `disabled`- when the provisioning network is fully disabled. User can bring up the
+                baremetal cluster using virtual media or assisted installation. If using metal3 for
+                power management, BMCs must be accessible from the machine networks. User should
+                provide two IPs on the external network that would be used for provisioning services.
+              enum:
+              - managed
+              - unmanaged
+              - disabled
```

Comment:
How are enums usually handled? The installer had assumed they were title-cased, which is what I saw elsewhere in the installer. I prefer lower case but I was just doing what I saw elsewhere

Reply:
That wan't clear to me either. The baremetalhost CRD usually used lower case values for its enum. So, I stuck with that.  In the MAO repo, the machineset CRD used title-cased enum values. I am not sure a standard exists.

Reply:
The constants are usually in the `UpperCamelCase` in openshift/api see for example https://github.com/openshift/api/pull/692#pullrequestreview-456364981

and that's what we try to follow in installer.